frontend/document/koptinterface.lua

```diff
@@ -968,10 +968,11 @@ function KoptInterface:getTextFromBoxes(boxes, pos0, pos1)
                         -- should be stuck
                         add_space = false
                     elseif dist_from_prev_word < box_height * 0.8 then
-                        if isCJKChar(prev_word:sub(-3, -1)) and isCJKChar(word:sub(1, 3)) then
-                            -- Two CJK chars whose spacing is not large enough
-                            -- (we checked the 3 UTF8 bytes that CJK chars must be,
-                            -- no need to split into unicode codepoints)
+                        local prev_word_end = prev_word:match(util.UTF8_CHAR_PATTERN.."$")
+                        local word_start = word:match(util.UTF8_CHAR_PATTERN)
+                        if util.isCJKChar(prev_word_end) and util.isCJKChar(word_start) then
+                            -- Two CJK chars whose spacing is not large enough,
```

Comment:
Anthropomorphizing the CJK characters? :-P

Reply:
Time for my English lesson :) How would you write this ?
I guess it's about `whose` ?
https://www.merriam-webster.com/words-at-play/whose-used-for-inanimate-objects
https://english.stackexchange.com/questions/23541/can-whose-refer-to-an-inanimate-object

(But in my childish heart, I consider CJK chars animals, like octopuses and ants, whose legs are many :)

Reply:
The other option would be "Two CJK chars which don't have a large enough spacing" or something like that. To me both sound okay, but "whose" is more concise which is why I wrote it that way :smile:. While "whose" does somewhat anthropomorphise them it didn't/doesn't sound particularly off to me.